### PR TITLE
refactor(zoe): severely restrict access to vatAdminSvc

### PR DESCRIPTION
**Pull request by katelynsills**
Attenuate `vatAdminSvc`. This will become more useful when the `startInstance` function is moved to another file. 

Includes changes from the PRs that were stacked on top of this, so there is less traffic going onto master:

https://github.com/Agoric/agoric-sdk/pull/3181
https://github.com/Agoric/agoric-sdk/pull/3182

### CODE REVIEW

File: packages/zoe/src/zoeService/createZCFVat.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * ZCF Vats can be created.
+ *
+ * @param {VatAdminSvc} vatAdminSvc
+ * @param {string=} zcfBundleName
+ * @returns {CreateZCFVat}
+ */
+export const setupCreateZCFVat = (vatAdminSvc, zcfBundleName) => {
+  /** @type {CreateZCFVat} */
+  const createZCFVat = () =>
+    zcfBundleName
```
Comment (erights) [inline packages/zoe/src/zoeService/createZCFVat.js]: When a parameter is optional I like to see it in its declaration as well.
```suggestion
export const setupCreateZCFVat = (vatAdminSvc, zcfBundleName = undefined) => {
  /** @type {CreateZCFVat} */
  const createZCFVat = () =>
    zcfBundleName !== undefined
```
or
```js
    typeof zcfBundleName === 'string'
```
depending on your intent, and on the requirements and assumptions of `createVatByName`.

File: packages/zoe/src/zoeService/zoe.js [hunk 2]
```diff
@@ -42,6 +42,11 @@ function makeZoe(vatAdminSvc, zcfBundleName = undefined) {
   /** @type {WeakStore<SeatHandle, ZoeSeatAdmin>} */
   const seatHandleToZoeSeatAdmin = makeNonVOWeakStore('seatHandle');
 
+  // This method contains the power to create a new ZCF Vat, and must
+  // be closely held. vatAdminSvc is even more powerful - any vat can
+  // be created. We severely restrict access to vatAdminSvc for this reason.
```
Comment (erights) [inline packages/zoe/src/zoeService/zoe.js:43]: Excellent

**Review by erights — APPROVED** [review]
LGTM

File: packages/zoe/src/zoeService/createZCFVat.js (hunk 1 above)
Comment (katelynsills) [reply packages/zoe/src/zoeService/createZCFVat.js]: Thanks for this. Done!

**Pull request merged**